### PR TITLE
Feature/948 document immutable component registry

### PR DESCRIPTION
**Pull request by weelink**
This PR for updating the documentation regarding the changes done for https://github.com/autofac/Autofac/pull/981 (making `ComponentRegistry` immutable).

I also removed references to updating the container.

### CODE REVIEW

File: docs/advanced/concurrency.rst [hunk 1]
```diff
@@ -7,9 +7,7 @@ Autofac is designed for use in highly-concurrent applications. The guidance belo
 Component Registration
 ----------------------
 
-``ContainerBuilder`` **is not thread-safe** and is designed to be used only on a single thread at the time the application starts up. This is the most common scenario and works for almost all applications.
-
-**Registration into a container** *after* **is is built, using** ``ContainerBuilder.Update()``, **also is not thread-safe.** For applications that register components after the container has been built (which should be very uncommon) additional locking to protect the container from concurrent access during an ``Update()`` operation is necessary.
+``ContainerBuilder`` and ``ComponentRegistryBuilder`` **are not thread-safe** and is designed to be used only on a single thread at the time the application starts up. This is the most common scenario and works for almost all applications.
```
Comment (tillig) [inline docs/advanced/concurrency.rst]: "...and are designed..." rather than "...and is designed..."

**Review by tillig — CHANGES_REQUESTED** [review]
Other than one minor change, this looks good. Thanks for following up here.

I guess I hadn't realized `ComponentRegistry` had become non-public. Good changes, to be sure, just something I hadn't noticed.

**Comment by tillig** [comment]
(Admittedly, we may hold off on merging this into the docs until we pull the Autofac `pr-immutable-container` branch into `develop`, but otherwise, yay!)

File: docs/advanced/concurrency.rst (hunk 1 above)
Comment (weelink) [reply docs/advanced/concurrency.rst]: Nice catch. I missed that one.

**Review by tillig — APPROVED** [review]
:tada:

**Comment by alexmg** [comment]
The `pr-immutable-container` branch has now been merged into `develop`. I think we should wait until the changes are available in a proper release before updating the documentation.

**Comment by tillig** [comment]
Ready to merge!

**Pull request merged**